sklearn/cluster/tests/test_optics.py

```diff
@@ -0,0 +1,342 @@
+# Authors: Shane Grigsby <refuge@rocktalus.com>
+#          Amy X. Zhang <axz@mit.edu>
+# License: BSD 3 clause
+
+import numpy as np
+
+from sklearn.datasets.samples_generator import make_blobs
+from sklearn.cluster.optics_ import OPTICS
+from sklearn.utils.testing import assert_equal, assert_warns
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_raises
+
+from sklearn.cluster.tests.common import generate_clustered_data
+
+
+def test_optics():
+    # Tests the optics clustering method and all functions inside it
+    # 'auto' mode
+
+    n_clusters = 3
+    X = generate_clustered_data(n_clusters=n_clusters)
+    print(np.shape(X))
+    # Parameters chosen specifically for this task.
+    # Compute OPTICS
+    clust = OPTICS(max_bound=5.0 * 6.0, min_samples=4, metric='euclidean')
+    clust.fit(X)
+    # number of clusters, ignoring noise if present
+    n_clusters_1 = len(set(clust.labels_)) - int(-1 in clust.labels_)
+    assert_equal(n_clusters_1, n_clusters)
+
+
+def test_optics2():
+    # Tests the optics clustering method and all functions inside it
+    # 'dbscan' mode
+
+    # Compute OPTICS
+    X = [[1, 1]]
+    clust = OPTICS(max_bound=5.0 * 0.3, min_samples=10)
+
+    # Run the fit
+    assert_raises(ValueError, clust.fit, X)
+
+
+def test_empty_extract():
+    # Test extract where fit() has not yet been run.
+    clust = OPTICS(max_bound=5.0 * 0.3, min_samples=10)
+    assert_raises(ValueError, clust.extract_dbscan, 0.01)
+
+
+def test_bad_extract():
+    # Test an extraction of eps too close to original eps
+    centers = [[1, 1], [-1, -1], [1, -1]]
+    X, labels_true = make_blobs(n_samples=750, centers=centers,
+                                cluster_std=0.4, random_state=0)
+
+    # Compute OPTICS
+    clust = OPTICS(max_bound=5.0 * 0.003, min_samples=10)
+    clust2 = clust.fit(X)
+    assert_raises(ValueError, clust2.extract_dbscan, 0.3)
+
+
+def test_close_extract():
+    # Test extract where extraction eps is close to scaled epsPrime
+
+    centers = [[1, 1], [-1, -1], [1, -1]]
+    X, labels_true = make_blobs(n_samples=750, centers=centers,
+                                cluster_std=0.4, random_state=0)
+
+    # Compute OPTICS
+    clust = OPTICS(max_bound=1.0, min_samples=10)
+    clust3 = clust.fit(X)
+    # check warning when centers are passed
+    assert_warns(RuntimeWarning, clust3.extract_dbscan, .3)
+    assert_equal(max(clust3.extract_dbscan(.3)[1]), 3)
+
+
+def test_auto_extract_hier():
+    # Generate sample data
+
+    np.random.seed(0)
```

Comment:
We prefer to avoid setting the seed globally in the tests.
Please use `rng = np.random.RandomState(0)` and `rng.randn(...)`.

You need to update your other tests too.

Reply:
Good to know-- fixed and updated for this and other tests.